packages/flutter_tools/lib/executable.dart

```diff
@@ -179,8 +179,11 @@ Future<Null> _exit(int code) async {
   logger.flush();
 
   // Give the task / timer queue one cycle through before we hard exit.
-  await Timer.run(() {
+  Completer<Null> completer = new Completer<Null>();
+  Timer.run(() {
```

Comment:
I think this'll do it:

```
  await new Future<Null>(() {
    printTrace('exiting with code $code');
    exit(code);
  });
```

So, no timer involved. The code's giving the tasks on the current task / timer queue a chance to run before hard-exiting the process. I'm afraid I don't remember the exact reason we're waiting here; perhaps to help the terminal output above write out?